src/file-upload/file-option-thumbnail.tsx

```diff
@@ -0,0 +1,29 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useEffect, useRef } from 'react';
+import styles from './styles.css.js';
+
+interface FileOptionThumbnailProps {
+  file: File;
+}
+
+export function FileOptionThumbnail({ file }: FileOptionThumbnailProps) {
+  const thumbnailRef = useRef<HTMLImageElement>(null);
+
+  useEffect(() => {
+    const reader = new FileReader();
+    reader.onloadend = () => {
+      if (thumbnailRef.current && thumbnailRef.current.src) {
+        thumbnailRef.current.src = reader.result as string;
```

Comment:
I would suggest limiting this by size, e.g. 50MB. Users might upload e.g. satellite images, which can be multi-gigabyte TIFF files. Loading these into a DOM attribute will likely slow down the page or device.

----

Can we also use React state for the `src` value instead of imperatively updating the DOM? Otherwise that state might get lost if/when React rerenders.

----

Potentially it might also be useful to use `URL.createObjectURL` instead of a FileReader, which will return a blob URL as a kind of "handle" to that file content. The advantage is that the (potentially large) file content will not be loaded into Javascript. The disadvantage is that this URL is not automatically garbage collected, so we'll have to call `revokeObjectURL` when the component unmounts or the file changes, to prevent a memory leak.